go/test/fuzzing/collate_fuzzer.go

```diff
@@ -0,0 +1,63 @@
+/*
+Copyright 2021 The Vitess Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package fuzzing
+
+import (
+	fuzz "github.com/AdaLogics/go-fuzz-headers"
+
+	"vitess.io/vitess/go/mysql/collations"
+)
+
+func FuzzCollateUnicode(data []byte) int {
+	c := &collations.Collation_unicode_general_ci{}
```

Comment:
This is not correct, you're not supposed to instantiate collation objects directly: they must be looked up directly from an Environment. I guess this is not properly explained in the API docs. :)

Reply:
Thanks for the heads up. I rewrote it a bit to initialize the same way as [the existing fuzzer](https://github.com/vitessio/vitess/blob/main/go/mysql/collations/uca_test.go#L53).